routing/payment_lifecycle_test.go

```diff
@@ -48,6 +51,110 @@ func createTestRoute(amt lnwire.MilliSatoshi,
 	)
 }
 
+// paymentLifecycleTestCase contains the steps that we expect for a payment
+// lifecycle test, and the routes that pathfinding should deliver.
+type paymentLifecycleTestCase struct {
```

Comment:
Nice refactor! I'm actually working on this file too, guess I'll rebase on your commit haha.